index.js

```diff
@@ -1,7 +1,53 @@
 // Iteration 1: Names and Input
 
+let hacker1 = "Miguel";
+let hacker2 = "Miguel";
 
-// Iteration 2: Conditionals
+console.log(`The driver´s name is ${hacker1}.`)
 
+console.log (`The navigator´s name is ${hacker2}.` )
 
-// Iteration 3: Loops
+//Iteration 2: Conditionals
+
+let size1 = hacker1.length ;
+let size2= hacker2.length;
+
+if ( size2 < size1) {
+    console.log (`The driver has the longest name, it has ${size1} characters.`);
+} else if ( size2 > size1) {
+    console.log(`It seems that the navigator has the longest name, it has ${size2} characters`);
+} else  {
+    console.log(`Wow, you both have equally long names, ${size2} characters!`);
+}
+
+//Iteration 3: Loops
+//3.1
+
+let driverName = hacker1.toUpperCase();
+let fdriver = "";
+
+for ( i= 0 ; i < size1 ; i++) {
+    if ( size1 - 1 === i) {
+        fdriver += driverName[i];
+    } else {
+        fdriver += driverName[i] + " ";
+    }
+}
+
+console.log(fdriver);
+
+//3.2
+
+    let reversed = "";
+    for (let i = hacker2.length - 1; i >= 0; i--) {
+      reversed += hacker2[i];
+    }
+
+//3.3
+if ( hacker1.localeCompare(hacker2) ) {
```

Comment:
The `localCompare()` method always returns 1, 0 or -1 so in this case it doesn't work as a condition.

- Negative when the `referenceStr` occurs before `compareString`
- Positive when the `referenceStr` occurs after `compareString`
- Returns 0 if they are equivalent

So, one of the ways you could have done this part is by storing the result of the `localCompare()` in a `variable`, and change the `conditions` to check if the result is positive negative or zero.

```
let compare = hacker1.localeCompare(hacker2)

if (compare === -1) {
  console.log("The driver name goes first");
} else if (compare === 1) {
  console.log("Yo, the navigator goes first definitely.");
} else {
  console.log("What?! You both have the same name?");
}
```